### DISCUSSION TITLE
style element with CSS variable being removed from DOM unexpected

### DISCUSSION THREAD

**Discussion by mellis481** [ant-design/ant-design#50279] [Q&A]
I am using Ant in each of the applications in my UI platform. Each application is implemented using a React-based microfrontend. At any given time, as few as 1 and as many as 3 microfrontends (each with their own instance of Ant) can be mounted/rendered on the page.

After upgrading to Ant 5, we found (https://ant.design/docs/react/css-variables#disable-hash) that we had to configure the `ConfigProvider` as follows to ensure multiple versions of Ant could exist on the page at one time without causing problems:

```
<ConfigProvider theme={{ cssVar: true, hashed: true }}>
  <App />
</ConfigProvider>
```

Everything has been working fine until recently when we noticed a problem. Here is what we're seeing:
- Load the page where 2-3 microfrontends are mounted.
- One of the microfrontends has button. The button `height` and `width` are set from a CSS in JS variable `var(--ant-control-height)`.
- This variable is defined in a `.css-var-r0` class which exists in a `<style data-rc-order="prependQueue" data-rc-priority="-999" data-css-hash="90g12z" data-token-hash="css-var-r0">` element.
- When I switch microfrontends (the one with the button remains on the page), that style element is removed from the DOM and the CSS variable becomes unavailable.

I noticed that if I changed the `data-css-hash` attribute of that `style` element after the initial load, the `style` block is not removed from the DOM when I switch microfrontends. This leads me believe that something inside of Ant is trying to clean up what it believes are unneeded `style` elements, but it's removing one that _is_ needed.

I'm working to create a reproduction of this, but any idea why this could be happening?

**Comment by HereSinceres** [comment]
step1:
`
  <ConfigProvider   theme={{cssVar: {key: 'theme'}}}></ConfigProvider>`
 step2:
`
 document.getElementsByTagName('html')[0].classList.add('theme');`
